errai-ioc/src/main/java/org/jboss/errai/ioc/client/api/EnabledByProperty.java

```diff
@@ -43,4 +43,15 @@
    * if the specified system property exists and has the value {@code "true"}.
    */
   boolean negated() default false;
+
+  /**
+   * Default value in case the system property is not set at all.
+   */
+  String defaultValue() default "";
```

Comment:
IMO this should be changed to a boolean because there are only two meaningful states:

1. Your `defaultValue` equals `matchValue`.
2. Your `defaultValue` does not equal `matchValue`.

I would suggest replacing this with

    boolean matchByDefault() default false;

where, when true, the default value of the system property will be the `matchValue` and `null` otherwise.